src/common/containers/landing.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+
+import { Anchor } from '../components/button';
+
+import styles from './container.css';
+
+class Landing extends Component {
+  componentWillMount() {
+    if (this.props.auth.authenticated) {
```

Comment:
Yes, redirect in component is a hack.

From the discussion here and in UX doc there seems to be preference for separate URL for dashboard, so it will probably end up like this.

Travis has dashboard on `/` and it works fine. Also doesn't really break the history (all other views have their URLs).

But I also agree with benefits of having access to landing page even if user is logged in. But still, these benefits are rather occasional (if they need to refer to some info there) and users that have account and are logged it are most likely to be interested in their dashboard and not landing page when they visit build.snapcraft.io.

Anyway. The URL decision needs to be taken outside of this PR anyway, but we can start with some URL and interate on that (as discussed in https://github.com/canonical-ols/build.snapcraft.io/pull/107#discussion_r100009669).

For the redirect I agree that it would be best to keep it server side.